### ISSUE TITLE
Quests: Add quest_ctx and rework tests to remove redundancy

### ISSUE THREAD

**Issue by JSerf02** [uchicago-cs/chiventure#1481]
When working on the many changes to the quest module, two issues became very clear: First, lots of functions need the same information but have them as different parameters with different names in different locations which is really confusing when you're trying to call the functions. Second, all of the tests require the same initialization and they are all copy pasted which pads out the test file and makes it unnecessarily hard to read.
We plan to fix the first issue by adding a quest_ctx class that can be passed as a parameter for every function so they all have the same info in the same place with the same name. Second, since we are going to need to update all of the tests anyways with the ctx changes, we plan to refactor all of the tests so they are far less redundant. This will likely include creating helper functions that initialize a game and a quest for testing.

**Comment by JSerf02** [comment]
Created the quest_ctx_t struct and added new, init, and free functions for it

**Comment by JSerf02** [comment]
We reworked all of the functions in the quest module that had player_t and quest_hash_t as parameters to use the new quest_ctx objects instead. We also adjusted the tests accordingly.

**Comment by JSerf02** [comment]
We refactored all of the tests from the quest module to make them stronger and remove redundant code through helper functions. Our tests are now stronger and way shorter and easier to read. While doing this, we encountered a few significant bugs and crashes that were extremely difficult to pin down. It's interesting to note that one of these bugs was actually a bug with a function from the player module that we happened to be calling on!
